torch/csrc/autograd/engine.cpp

```diff
@@ -858,33 +927,45 @@ bool Engine::is_checkpoint_valid() {
   return checkpoint_valid;
 }
 
-size_t Engine::ready_queue_size(at::Device device) {
+size_t Engine::ready_queue_size(const std::shared_ptr<GraphTask>& graph_task, at::Device device) {
   if (ready_queues_.empty()) {
     // The vector ready_queues_ is initialized in start_threads, but this method
     // can be called before start_threads. Adding this check to avoid index
     // out of bound error.
     return 0;
   }
-  return ready_queue(device).size();
+  return ready_queue(graph_task, device).size();
 }
 
-auto Engine::ready_queue(at::Device device) -> ReadyQueue& {
-  // See Note [Allocating GPUs to autograd threads]
+auto Engine::ready_queue(const std::shared_ptr<GraphTask>& graph_task, at::Device device) -> ReadyQueue& {
```

Comment:
Probably good to have a comment here, something like "CPU queue is per GraphTask, but CUDA device queues are shared across all graph tasks"